scripts/fast_lint.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python3
+
+"""Runs the common linters on any files that have changed relative to the main branch."""
+from __future__ import annotations
+
+import argparse
+import concurrent.futures
+import logging
+import os
+import subprocess
+import sys
+import time
+
+from git import Repo
+
+
+def changed_files() -> list[str]:
+    repo = Repo(os.getcwd())
+
+    current_branch = repo.active_branch
+    common_ancestor = repo.merge_base(current_branch, "main")[0]
+
+    return [item.b_path for item in repo.index.diff(common_ancestor)]
+
+
+def run_cmd(cmd: list[str], files: list[str] | bool) -> bool:
+    start = time.time()
+    if not files:
```

Comment:
This becomes `if files is not None and len(files) == 0:`. Verbose but arguably more explicit on the input semantic.